eth_abi/decoding.py

```diff
@@ -548,18 +547,4 @@ def from_type_str(cls, abi_type, registry):
 
     @staticmethod
     def decoder_fn(data):
-        try:
-            value = data.decode("utf-8")
-        except UnicodeDecodeError as e:
-            raise DecodingError(
-                e.encoding,
-                e.object,
-                e.start,
-                e.end,
-                "The returned type for this function is string which is "
-                "expected to be a UTF8 encoded string of text. The returned "
-                "value could not be decoded as valid UTF8. This is indicative "
-                "of a broken application which is using incorrect return types for "
-                "binary data.",
-            ) from e
-        return value
+        return data.decode("utf-8", errors="surrogateescape")
```

Comment:
`decode` can take a number of different error callbacks for strings: `"strict"`, `"ignore"`, `"replace"`, `"surrogateescape"`, `"backslashreplace"`, and `"xmlcharrefreplace"`. We can allow users to use any of these callbacks if we define our own `__call__` and `decode` functions inside the `StringDecoder` class that accept an error handler. What do you think about that? 

So the class would look something like:

```
class StringDecoder(ByteStringDecoder):
    def __call__(self, stream: ContextFramesBytesIO, errors="surrogateescape") -> Any:
        return self.decode(stream, errors)

    @parse_type_str("string")
    def from_type_str(cls, abi_type, registry):
        return cls()

    def decode(self, stream, errors="surrogateescape"):
        raw_data = self.read_data_from_stream(stream)
        data, padding_bytes = self.split_data_and_padding(raw_data)
        value = data.decode("utf-8", errors)
        self.validate_padding_bytes(value, padding_bytes)

        return value
```